libasynql/src/poggit/libasynql/mysqli/MysqliThread.php

```diff
@@ -42,28 +43,38 @@
 use function assert;
 use function bccomp;
 use function bcsub;
+use function count;
 use function gettype;
 use function implode;
 use function in_array;
 use function is_float;
 use function is_int;
 use function is_string;
+use function min;
 use function serialize;
+use function sleep;
 use function strtotime;
 use function unserialize;
+use const PHP_INT_MAX;
 
 class MysqliThread extends SqlSlaveThread{
 	/** @var string */
 	private $credentials;
+	/** @var string */
+	private $logger;
+	/** @var int */
+	private $attempts;
```

Comment:
Is it necessary to store this as a class property? It seems you are only using it as a local variable.